src/Data/ProgrammesDb/EntityRepository/PromotionRepository.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository;
+
+use BBC\ProgrammesPagesService\Domain\ValueObject\Pid;
+use DateTimeImmutable;
+use Doctrine\ORM\EntityRepository;
+use Doctrine\ORM\Query;
+
+class PromotionRepository extends EntityRepository
+{
+    use Traits\ParentTreeWalkerTrait;
+
+    /**
+     * @return array[]
+     */
+    public function findActivePromotionsByPid(Pid $pid, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+            ->addSelect(['promotionOfCoreEntity', 'promotionOfImage', 'imageForPromotionOfCoreEntity'])
+            ->innerJoin('promotion.context', 'context')
+            ->leftJoin('promotion.promotionOfCoreEntity', 'promotionOfCoreEntity')
+            ->leftJoin('promotion.promotionOfImage', 'promotionOfImage')
+            ->leftJoin('promotionOfCoreEntity.image', 'imageForPromotionOfCoreEntity')
+            ->andWhere('promotion.isActive = 1')
+            ->andWhere('promotion.startDate <= :datetime')
+            ->andWhere('promotion.endDate >= :datetime')
+            ->andWhere('context.pid = :pid')
+            ->addOrderBy('promotion.weighting', 'ASC')
+            ->setFirstResult($offset)
+            ->setMaxResults($limit)
+            ->setParameter('pid', $pid)
+            ->setParameter('datetime', $datetime);
+
+        $promotions = $qb->getQuery()->getResult(Query::HYDRATE_ARRAY);
+        return $this->resolveParentsForPromosOfCoreEntities($promotions);
+    }
+
+    /**
+     * @param int[] $ancestryIds
+     * @return array[]
+     */
+    public function findActiveSuperPromotionsByPid(array $ancestryIds, DateTimeImmutable $datetime, ?int $limit, int $offset): array
+    {
+        $qb = $this->createQueryBuilder('promotion')
+           ->addSelect(['promotionOfCoreEntity', 'promotionOfImage', 'imageForPromotionOfCoreEntity'])
+           ->innerJoin('promotion.context', 'context')
+           ->leftJoin('promotion.promotionOfCoreEntity', 'promotionOfCoreEntity')
+           ->leftJoin('promotion.promotionOfImage', 'promotionOfImage')
+           ->leftJoin('promotionOfCoreEntity.image', 'imageForPromotionOfCoreEntity')
+           ->andWhere('promotion.isActive = 1')
+           ->andWhere('promotion.startDate <= :datetime')
+           ->andWhere('promotion.endDate >= :datetime')
```

Comment:
Should this be `promotion.endDate > :datetime`? In most other entities that have an 'endDate' (e.g. Service) we do `>`


IF something ends at 2017-08-08 12:00:00` do